ui/store/actions.js

```diff
@@ -624,196 +625,36 @@ export function signTypedMsg(msgData) {
-        dispatch(gasLoadingFinished());
-      });
-  };
-}
-
-export function gasLoadingStarted() {
-  return {
-    type: actionConstants.GAS_LOADING_STARTED,
-  };
-}
-
-export function gasLoadingFinished() {
-  return {
-    type: actionConstants.GAS_LOADING_FINISHED,
-  };
-}
-
-export function updateSendTokenBalance({ sendToken, tokenContract, address }) {
-  return (dispatch) => {
-    const tokenBalancePromise = tokenContract
-      ? tokenContract.balanceOf(address)
-      : Promise.resolve();
-    return tokenBalancePromise
-      .then((usersToken) => {
-        if (usersToken) {
-          const newTokenBalance = calcTokenBalance({ sendToken, usersToken });
-          dispatch(setSendTokenBalance(newTokenBalance));
-        }
-      })
-      .catch((err) => {
-        log.error(err);
-        updateSendErrors({ tokenBalance: 'tokenBalanceError' });
-      });
-  };
-}
-
-export function updateSendErrors(errorObject) {
-  return {
-    type: actionConstants.UPDATE_SEND_ERRORS,
-    value: errorObject,
-  };
-}
-
-export function setSendTokenBalance(tokenBalance) {
-  return {
-    type: actionConstants.UPDATE_SEND_TOKEN_BALANCE,
-    value: tokenBalance,
-  };
-}
-
-export function updateSendHexData(value) {
-  return {
-    type: actionConstants.UPDATE_SEND_HEX_DATA,
-    value,
-  };
-}
-
-export function updateSendTo(to, nickname = '') {
-  return {
-    type: actionConstants.UPDATE_SEND_TO,
-    value: { to, nickname },
-  };
-}
-
-export function updateSendAmount(amount) {
-  return {
-    type: actionConstants.UPDATE_SEND_AMOUNT,
-    value: amount,
-  };
-}
-
 export function updateCustomNonce(value) {
   return {
     type: actionConstants.UPDATE_CUSTOM_NONCE,
     value,
   };
 }
 
-export function setMaxModeTo(bool) {
-  return {
-    type: actionConstants.UPDATE_MAX_MODE,
-    value: bool,
-  };
-}
-
-export function updateSend(newSend) {
-  return {
-    type: actionConstants.UPDATE_SEND,
-    value: newSend,
-  };
-}
-
-export function updateSendToken(token) {
-  return {
-    type: actionConstants.UPDATE_SEND_TOKEN,
-    value: token,
-  };
-}
-
-export function clearSend() {
-  return {
-    type: actionConstants.CLEAR_SEND,
-  };
-}
-
-export function updateSendEnsResolution(ensResolution) {
-  return {
-    type: actionConstants.UPDATE_SEND_ENS_RESOLUTION,
-    payload: ensResolution,
-  };
-}
-
-export function updateSendEnsResolutionError(errorMessage) {
-  return {
-    type: actionConstants.UPDATE_SEND_ENS_RESOLUTION_ERROR,
-    payload: errorMessage,
-  };
-}
-
 export function signTokenTx(tokenAddress, toAddress, amount, txData) {
   return async (dispatch) => {
     dispatch(showLoadingIndication());
 
     try {
       const token = global.eth.contract(abi).at(tokenAddress);
-      const txPromise = token.transfer(toAddress, addHexPrefix(amount), txData);
-      dispatch(showConfTxPage());
+      token.transfer(toAddress, addHexPrefix(amount), txData);
+      await wait(500);
```

Comment:
This wait is necessary because there is no guarantee that after the `token.transfer` call is made that the transaction has fully made its way into the unapproved transaction state. I am not sure why moving the send state out of the metamask state makes this issue happen but the way it sufficient to rectify it. Tried with 200,300 and 400.500 is successful 100% of the time while 400 had intermittent test failures.